src/fshtypes/FshEntity.ts

```diff
@@ -0,0 +1,13 @@
+/**
+ * Abstract class to contain elements common to all FSH types.
+ */
+export abstract class FshEntity {
+  constructor(public location?: TextLocation) {}
+}
+
+export type TextLocation = {
```

Comment:
I'm also wondering what you think about naming this variable `source` instead of `location`?  I'm not asking you to rename it -- I'm really just asking for your honest opinion.  My thinking is that if we want to add more info in the future (like maybe the actual FSH text string), then `location` might be too specific?

Reply:
Tell me if this sounds okay: `location` stays as it is, and `FshEntity` now has a new string attribute `source` that has the name of the file (empty if the text was read from a string/stdin)

Reply:
That's probably fine, but I was thinking they kind of go together -- because line and column are kind of meaningless without the source file reference.  And thinking ahead, if we do decide to include the text of the line (so the logger can output that in verbose mode), then do we add another top-level attribute to FshEntity for that?  I feel like since they are all for a similar purpose, they should be grouped.

Another alternative:
```ts
export abstract class FshEntity {
  source: SourceInfo;
}

type SourceInfo = {
  file: string,
  location: TextLocation,
  text: string
}
```
(we wouldn't add `text` now, but showing it for later)...  Maybe it should even be a class (so we can have a constructor)?  Not sure -- I'm still learning when it's best to use a `class` vs `type` vs `interface`.

Reply:
We could also call it `debug` and `DebugInfo` if we want to even more broad.